### PR TITLE
Update to gradle 8

### LINKED ISSUE #136: Update to gradle 8

### PR DESCRIPTION
**Pull request by danielfeismann**
resolves #136

### CODE REVIEW

**Comment by danielfeismann** [comment]
!test

File: build.gradle [hunk 1]
```diff
@@ -89,7 +91,7 @@ dependencies {
 compileScala {
   scalaCompileOptions.additionalParameters = [
     "-Xplugin:" + configurations.scalaCompilerPlugin.asPath,
-    "-P:scapegoat:dataDir:" + buildDir + "/reports/scapegoat/src/",
+    "-P:scapegoat:dataDir:" + rootDir + "/build/reports/scapegoat/src/",
```
Comment (sebastian-peter) [inline build.gradle:94]: One can use this instead to reference the build directory in gradle (same below):

```suggestion
    "-P:scapegoat:dataDir:" + layout.buildDirectory.dir("reports/scapegoat/src/"),
```

File: gradle/scripts/mavenCentralPublish.gradle [hunk 2]
```diff
@@ -88,7 +88,7 @@ if (project.hasProperty('user') && project.hasProperty('password')) {
 
   model {
     tasks.generatePomFileForMavenJavaPublication {
-      destination = file("$buildDir/generated-pom.xml")
+      destination = file("$rootDir/build/generated-pom.xml")
```
Comment (sebastian-peter) [inline gradle/scripts/mavenCentralPublish.gradle:91]: Same as above (if applicable)

**Review by sebastian-peter — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (sebastian-peter) [reply build.gradle:94]: Then nevermind

**Review by sebastian-peter — APPROVED** [review]

**Pull request merged**